### PR TITLE
Implement logic for EntityMock, LivingEntityMock, ArmorStandMock, HumanEntityMock and PlayerMock

### PR DESCRIPTION
**Pull request by 4everTheOne**
# Description
This PR implements logic for the classes EntityMock, LivingEntityMock, ArmorStandMock, HumanEntityMock and PlayerMock.
Initially it was started with the intention to update only ArmorStandMock, but since it has functions in command with EntityMock and LivingEntityMock it was required to also update them. 

The classes HumanEntityMock and PlayerMock was done a cleanup do remove redudant function that are now present in the LivingEntityMock or EntityMock.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [x] Unit tests added.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/entity/ArmorStandMock.java [hunk 1]
```diff
@@ -386,85 +395,97 @@ public boolean isSlotDisabled(@NotNull EquipmentSlot slot)
...
+		return toPaper(getBodyPose());
 	}
 
 	@Override
 	public void setBodyRotations(@NotNull Rotations rotations)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkNotNull(rotations, "Rotations cannot be null");
+		setBodyPose(toBukkit(rotations));
```
Comment (4everTheOne) [inline src/main/java/be/seeseemelk/mockbukkit/entity/ArmorStandMock.java:407]: This was based on the Paper implementation and in paper implementation, both "setBodyRotations" and "setBodyPose" share the same variable.

File: src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java [hunk 2]
```diff
@@ -1171,8 +1182,14 @@ public boolean removeScoreboardTag(@NotNull String tag)
 	@Override
 	public void setRotation(float yaw, float pitch)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		NumberConversions.checkFinite(pitch, "pitch not finite");
+		NumberConversions.checkFinite(yaw, "yaw not finite");
+
+		yaw = Location.normalizeYaw(yaw);
+		pitch = Location.normalizePitch(pitch);
```
Comment (4everTheOne) [inline src/main/java/be/seeseemelk/mockbukkit/entity/EntityMock.java:1189]: Based on paper implementation.

**Comment by Insprill** [comment]
This should be rebased onto the `update-1.21` branch. Hopefully there isn't too many conflicts 😅

**Comment by 4everTheOne** [comment]
Thank you @Insprill for the information. I have updated the PR to be compatible with v1.21

**Pull request closed without merging**